src/cards/polymorph.ts

```diff
@@ -0,0 +1,194 @@
+      return undefined;
+    }
+  }
+  const toSourceUnit = allUnits[toUnitId];
+  if (!toSourceUnit) {
+    console.error('Unit with id', toUnitId, 'does not exist. Have you registered it in src/units/index.ts?');
+    return undefined;
+  }
+
+  // Cases for polymorphing AI / Player
+  if (fromUnit.unitType != UnitType.PLAYER_CONTROLLED) {
+    let unit: Unit.IUnit = Unit.create(
+      toSourceUnit.id,
+      fromUnit.x,
+      fromUnit.y,
+      fromUnit.faction,
+      toSourceUnit.info.image,
+      UnitType.AI,
+      toSourceUnit.info.subtype,
+      { ...toSourceUnit.unitProps, isMiniboss: fromUnit.isMiniboss, originalLife: fromUnit.originalLife },
+      underworld,
+      prediction
+    );
+
+    if (unit != undefined) {
+      // Keep Modifiers
+      for (const modifierKey of Object.keys(unit.modifiers)) {
+        const modifier = allModifiers[modifierKey];
+        const modifierInstance = unit.modifiers[modifierKey];
+        if (modifier && modifierInstance) {
+          if (modifier?.add) {
+            modifier.add(unit, underworld, prediction, modifierInstance.quantity, modifierInstance);
+          }
+        } else {
+          console.error("Modifier doesn't exist? This shouldn't happen.");
+        }
+      }
+
+      // Cleanup old unit and remove it from targets
+      Unit.cleanup(fromUnit, false);
+    }
+    return unit;
+  } else {
+    // Only change vfx/sfx for player
+    fromUnit.defaultImagePath = toSourceUnit.unitProps.defaultImagePath || fromUnit.defaultImagePath;
+    fromUnit.animations = toSourceUnit.animations || fromUnit.animations;
+    fromUnit.sfx = toSourceUnit.sfx || fromUnit.sfx;
+    fromUnit.bloodColor = toSourceUnit.unitProps.bloodColor || fromUnit.bloodColor;
+
+    Image.changeSprite(
+      fromUnit.image,
+      fromUnit.animations.idle,
+      fromUnit.image?.sprite.parent,
+      undefined,
+    );
+
+    return fromUnit;
+  }
+}
+
+function polymorphPickup(fromPickup: IPickup, underworld: Underworld, prediction: boolean, toPickupSource?: Pickup.IPickupSource): IPickup | undefined {
+  // If a specific ID isn't passed in, choose a random one
+  if (!toPickupSource) {
+    // Don't polymorph purple portals
+    if (fromPickup.name == Pickup.PORTAL_PURPLE_NAME) return undefined;
+
+    let possiblePickupTypes = pickups.filter(p => isModActive(p, underworld) && p.name != fromPickup.name
+      && p.name != Pickup.PORTAL_PURPLE_NAME && p.name != Pickup.RECALL_POINT);
+
+    // We have to seed this to prevent multiplayer desync
+    // ID is required here to ensure that every polymorph changes the seed to prevent looping
+    const seed = seedrandom(`${getUniqueSeedString(underworld)} - ${fromPickup.id}`);
+    toPickupSource = chooseOneOfSeeded(possiblePickupTypes, seed);
+    if (toPickupSource == undefined) {
+      console.error("Polymorph failed to choose a new pickup type.");
+      return undefined;
+    }
+  }
+
+  console.log(toPickupSource);
```

Comment:
remove log.

Tip: When i'm testing stuff while I develop with logs I usually put a unique identifier at the front or end such as 
`console.log('jtest', toPickupSource)`.  I do this because
1. It's easy to find them all later and remove the test ones with a simple search
2. If one gets left in accidentally it's easy to find and remove.

I've worked on many projects where a testing log accidentally gets committed and then it's really hard to find because it's logging just an object and so is rather unsearchable in the codebase.